### PR TITLE
fix(Core/Item): Implement mount speed mod items

### PR DESCRIPTION
**Pull request by Kitzunu**
## Changes Proposed:
-  Implement Riding Crop / Skybreaker Whip
-  Group these items together in one script
- Riding Crop / Skybreaker Whip SHOULD NOT stack with any other item
- Carrot on a stick + Mithril Spurs SHOULD stack

## Tests Performed:
- Builds
- Tested in-game
image
image
image
image

## How to Test the Changes:

Make sure to be <= lvl 70
1. .learn 61425 (Tundra Mammoth alliance)
2. .add 32863/25653/11122/7969

## Target Branch(es):
- [x] Master

 
## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

File: data/sql/updates/pending_db_world/rev_1610385898427311800.sql [hunk 1]
```diff
@@ -0,0 +1,8 @@
+INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1610385898427311800');
+
+DELETE FROM `spell_script_names` WHERE `spell_id` IN (48777,7215,48776,25653);
+INSERT INTO `spell_script_names` (`spell_id`, `ScriptName`) VALUES
+(48777,'spell_item_with_mount_speed'),
+(7215,'spell_item_with_mount_speed'),
+(48776,'spell_item_with_mount_speed'),
+(25653, 'spell_item_with_mount_speed');
```
Comment (pklloveyou) [inline data/sql/updates/pending_db_world/rev_1610385898427311800.sql:8]: del ```(25653, 'spell_item_with_mount_speed');```

File: data/sql/updates/pending_db_world/rev_1610385898427311800.sql [hunk 2]
```diff
@@ -0,0 +1,8 @@
+INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1610385898427311800');
+
+DELETE FROM `spell_script_names` WHERE `spell_id` IN (48777,7215,48776,25653);
```
Comment (pklloveyou) [inline data/sql/updates/pending_db_world/rev_1610385898427311800.sql:3]: del ```25653```

**Comment by pklloveyou** [comment]
```
DELETE FROM `spell_script_names` WHERE `spell_id` IN (48777,7215,48776);
INSERT INTO `spell_script_names` (`spell_id`, `ScriptName`) VALUES
(48777,'spell_item_with_mount_speed'),
(7215,'spell_item_with_mount_speed'),
(48776,'spell_item_with_mount_speed');
```

File: data/sql/updates/pending_db_world/rev_1610385898427311800.sql (hunk 1 above)
Comment (Kitzunu) [reply data/sql/updates/pending_db_world/rev_1610385898427311800.sql:8]: Ah yes, silly misstake

**Comment by pklloveyou** [comment]
tested

File: src/server/scripts/Spells/spell_item.cpp [hunk 3]
```diff
@@ -258,43 +258,87 @@ class spell_item_goblin_gumbo_kettle : public SpellScriptLoader
...
+                    break;
+                case SPELL_MITHRIL_SPURS:
+                    spellId = SPELL_MITHRIL_SPURS_EFFECT;
+                    break;
+                case SPELL_MOUNT_SPEED_RIDING:
+                    spellId = SPELL_RIDING_CROP_EFFECT;
+                    break;
+                default:
+                    return;
+                }
```
Comment (FrancescoBorzi) [inline src/server/scripts/Spells/spell_item.cpp:303]: indentation

File: src/server/scripts/Spells/spell_item.cpp [hunk 4]
```diff
@@ -258,43 +258,87 @@ class spell_item_goblin_gumbo_kettle : public SpellScriptLoader
...
+        PrepareAuraScript(spell_item_with_mount_speed_AuraScript);
 
-        void OnApply(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
+        void OnApply(AuraEffect const* aurEff, AuraEffectHandleModes /*mode*/)
         {
             Unit* target = GetTarget();
             if (target->getLevel() <= 70)
-                target->AddAura(SPELL_MITHRIL_SPEED, target);
+            {
+                uint32 spellId = 0;
```
Comment (FrancescoBorzi) [inline src/server/scripts/Spells/spell_item.cpp:288]: isn't `uint16` enough for this?

File: src/server/scripts/Spells/spell_item.cpp [hunk 5]
```diff
@@ -258,43 +258,87 @@ class spell_item_goblin_gumbo_kettle : public SpellScriptLoader
...
+
+                target->CastSpell(target, spellId, aurEff);
+            }
         }
 
         void OnRemove(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
         {
             Unit* target = GetTarget();
-            target->RemoveAurasDueToSpell(SPELL_MITHRIL_SPEED);
+            uint32 spellId = 0;
```
Comment (FrancescoBorzi) [inline src/server/scripts/Spells/spell_item.cpp:312]: same as above

File: src/server/scripts/Spells/spell_item.cpp [hunk 6]
```diff
@@ -258,43 +258,87 @@ class spell_item_goblin_gumbo_kettle : public SpellScriptLoader
...
+                break;
+            case SPELL_MITHRIL_SPURS:
+                spellId = SPELL_MITHRIL_SPURS_EFFECT;
+                break;
+            case SPELL_MOUNT_SPEED_RIDING:
+                spellId = SPELL_RIDING_CROP_EFFECT;
+                break;
+            default:
+                return;
+            }
```
Comment (FrancescoBorzi) [inline src/server/scripts/Spells/spell_item.cpp:327]: this part is duplicate, so why not wrapping this into a method instead?

Comment (Kitzunu) [reply src/server/scripts/Spells/spell_item.cpp:327]: Cuz I dont know how to do that

Comment (FrancescoBorzi) [reply src/server/scripts/Spells/spell_item.cpp:327]: I can help you later, please remind me

Comment (FrancescoBorzi) [reply src/server/scripts/Spells/spell_item.cpp:327]: @Kitzunu see if the following works for you: https://github.com/Kitzunu/azerothcore-wotlk/pull/2

Comment (Kitzunu) [reply src/server/scripts/Spells/spell_item.cpp:322]: @FrancescoBorzi this does not work

Comment (Kitzunu) [reply src/server/scripts/Spells/spell_item.cpp:322]: Yep this works =) Removed the "this" as it was not neccessary

**Comment by pklloveyou** [comment]
tested

**Review by FrancescoBorzi — APPROVED** [review]

**Pull request merged**